Homework_12/ValidatorTest.java

```diff
@@ -3,80 +3,102 @@
 import org.testng.annotations.Test;
 import org.testng.annotations.BeforeTest;
 import org.testng.annotations.DataProvider;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.File;
+
 
 /**
  * Created by Nick on 07/11/16.
  */
 public class ValidatorTest {
     private Solve solve;
+    private static String FilePath = "./unitTestInputData.xml";
 
     @BeforeTest
     public void setUp() throws Exception {
         solve = new Solve();
     }
 
-    @DataProvider(name = "positiveForEquilateral")
-    public Object[][] getNumber() {
-        return new Object[][]{
-                {1.0, 1.0, 1.0},
-                {Double.MAX_VALUE, Double.MAX_VALUE, Double.MAX_VALUE},
-                {10.0, 10.0, 10.0},
-                {156.3, 156.3, 156.3},
-                {1765.67, 1765.67, 1765.67},
-        };
+    public static Object[][] readFromXML(String tagName) throws Exception {
+        File inputFile = new File(FilePath);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
I've asked you to move this common code to separate entity.... but you just ignored it. It makes me sad